recipes/corrade/all/conanfile.py

```diff
@@ -97,54 +101,59 @@ def package(self):
 
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
-        self.env_info.PATH.append(bindir)
+        # The FindCorrade.cmake file provided by the library populates some extra stuff
+        self.cpp_info.components["_corrade"].build_modules.append(os.path.join("lib", "cmake", "conan-corrade-vars.cmake"))
+
+        if self.options.with_main:
+            self.cpp_info.components["main"].names["cmake_find_package"] = "Main"
+            self.cpp_info.components["main"].names["cmake_find_package_multi"] = "Main"
+            if self.settings.os == "Windows":
+                self.cpp_info.components["main"].libs = ["CorradeMain" + suffix]
+            self.cpp_info.components["main"].requires = ["_corrade"]
+
+        if self.options.with_utility:
+            self.cpp_info.components["utility"].names["cmake_find_package"] = "Utility"
+            self.cpp_info.components["utility"].names["cmake_find_package_multi"] = "Utility"
+            self.cpp_info.components["utility"].libs = ["CorradeUtility" + suffix]
+            if self.settings.os == "Linux":
+                self.cpp_info.components["utility"].system_libs = ["m", "dl"]
+            self.cpp_info.components["utility"].requires = ["_corrade"]
+
+            # This one is statically linked into utility
+            #self.cpp_info.components["containers"].names["cmake_find_package"] = "Containers"
+            #self.cpp_info.components["containers"].names["cmake_find_package_multi"] = "Containers"
+            #self.cpp_info.components["containers"].libs = ["CorradeContainers" + suffix]
+
+        if self.options.with_interconnect:
+            self.cpp_info.components["interconnect"].names["cmake_find_package"] = "Interconnect"
+            self.cpp_info.components["interconnect"].names["cmake_find_package_multi"] = "Interconnect"
+            self.cpp_info.components["interconnect"].libs = ["CorradeInterconnect" + suffix]
+            self.cpp_info.components["interconnect"].requires = ["utility"]
+
+        if self.options.with_pluginmanager:
+            self.cpp_info.components["plugin_manager"].names["cmake_find_package"] = "PluginManager"
+            self.cpp_info.components["plugin_manager"].names["cmake_find_package_multi"] = "PluginManager"
+            self.cpp_info.components["plugin_manager"].libs = ["CorradePluginManager" + suffix]
+            self.cpp_info.components["plugin_manager"].requires = ["utility"]
+
+        if self.options.with_testsuite:
+            self.cpp_info.components["test_suite"].names["cmake_find_package"] = "TestSuite"
+            self.cpp_info.components["test_suite"].names["cmake_find_package_multi"] = "TestSuite"
+            self.cpp_info.components["test_suite"].libs = ["CorradeTestSuite" + suffix]
+            self.cpp_info.components["test_suite"].requires = ["utility"]
```

Comment:
Too generic conan component names, it might conflicts with others recipes if consumer use pkg_config generator.

Reply:
yes, that's good hint, most people (including myself) don't realize there is name clashing problem for certain generators....

Reply:
I added these lines at the end of the `package_info` method, it prepends `corrade_` to all the components

```python
        # pkg_config: Add more explicit naming to generated files (avoid filesystem collision).
        for key, cmp in self.cpp_info.components.items():
            self.cpp_info.components[key].names["pkg_config"] = "{}_{}".format(self.name, key)
```